cmd/job_run_list.go

```diff
@@ -0,0 +1,156 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/odpf/salt/log"
+	cli "github.com/spf13/cobra"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	pb "github.com/odpf/optimus/api/proto/odpf/optimus/core/v1beta1"
+)
+
+const (
+	jobStatusTimeout = time.Second * 30
+)
+
+type jobRunCmdArg struct {
+	projectName string
+	jobName     string
+	startDate   string
+	endDate     string
+}
+
+type dateRange struct {
+	start time.Time
+	end   time.Time
+}
+
+type getRequestFn func(jobRunCmdArg) (*pb.JobRunRequest, error)
+
+func jobRunListCommand(l log.Logger, defaultProjectName, defaultHost string) *cli.Command {
+	cmd := &cli.Command{
+		Use:     "runs",
+		Short:   "Get current job runs",
+		Example: `optimus job runs <sample_job_goes_here> [--project \"project-id\"] [--start_date \"2006-01-02T15:04:05Z07:00\" --end_date \"2006-01-02T15:04:05Z07:00\"]`,
+		Args:    cli.MinimumNArgs(1),
+	}
+	projectName := defaultProjectName
+	host := defaultHost
+	var startDate string
+	var endDate string
+	cmd.Flags().StringVarP(&projectName, "project", "p", projectName, "Project name of optimus managed repository")
+	cmd.Flags().StringVar(&host, "host", defaultHost, "Optimus service endpoint url")
+	cmd.Flags().StringVar(&startDate, "start_date", "", "start date of job run")
+	cmd.Flags().StringVar(&endDate, "end_date", "", "end date of job run")
+	cmd.RunE = func(c *cli.Command, args []string) error {
+		jobName := args[0]
+		l.Info(fmt.Sprintf("Requesting status for project %s, job %s from %s",
+			projectName, jobName, host))
+		arg := jobRunCmdArg{
```

Comment:
instead of creating the jobRunCmdArg and then the jobRunRequest and doing the validation somewhere down the line. Lets add just three functions.
1. One for validation.
2. One for requestCreation.
3. getJobRunList

We would not need anymore functions.

Reply:
1. removed `jobRunCmdArg`
2. refactor `getJobRunList` to validate the cmd args, creating request and getting the response.
3. removed `getRequestFn` type 
4. removed `getJobRunRequest` function
5. removed `getJobRunRequestWithDate` function
6. simplify validate args function and renamed to `validateDateArgs `
7. handled request creation under single function `createJobRunRequest`